File: Sum.m
classdef Sum < Module
    
    properties(SetAccess=private)    
        output
        numberOfInputs
    end
        
    methods
       
        function this = Sum(name, numberOfInputs)
           
            this = this@Module(name);
            
            for n = 1:numberOfInputs
                
                this.createInputPort();
                
            end;
           
            this.numberOfInputs = numberOfInputs;
            this.output = this.createOutputPort();
        end
        
        function doUpdate(this,N)
           
            y = 0;
            
            for n = 1:this.numberOfInputs
                
                y = y + this.inputPorts(n).read(N);
                
            end;
            
            this.output.write( y );
        end
    end
end

File: Cpp/Modules/Source/Sum.cpp
#include "Sum.h"

Sum::Sum(size_t numberOfInputs)
{
	for(size_t n = 0; n < numberOfInputs; ++n)
	{
		CreateInputPort();
	}

	_pOutput = CreateOutputPort();
}

void Sum::Update()
{
	double y = 0.0;

	for (size_t n = 0; n < GetNumberOfInputs(); ++n)
	{
		y += GetInputPort(n)->Read();
	}

	_pOutput->Write(y);
}
	
IOutputPort* Sum::Output() const
{
	return _pOutput;
}
